jfrog-cli/artifactory/utils/golang/project/dependencies/dependenciesutils.go

```diff
@@ -0,0 +1,452 @@
+			if i < len(moduleName)-1 {
+				r := rune(moduleName[i+1])
+				str += string(unicode.ToUpper(r))
+				i++
+			}
+		} else {
+			str += string(moduleName[i])
+		}
+	}
+	return str
+}
+
+// Runs the go mod download command. Should set first the environment variable of GoProxy
+func downloadDependency(downloadFromArtifactory bool, fullDependencyName, targetRepo string, details *config.ArtifactoryDetails) error {
+	var err error
+	if downloadFromArtifactory {
+		log.Debug("Downloading dependency from Artifactory:", fullDependencyName)
+		err = golang.SetGoProxyEnvVar(details, targetRepo)
+	} else {
+		log.Debug("Downloading dependency from VCS:", fullDependencyName)
+		err = os.Unsetenv(golang.GOPROXY)
+	}
+	if errorutils.CheckError(err) != nil {
+		return err
+	}
+
+	err = golang.DownloadDependency(fullDependencyName)
+	return errorutils.CheckError(err)
+}
+
+func populateModAndGetDependenciesGraph(path string, shouldRunGoModCommand, shouldRunGoGraph bool) (output map[string]bool, err error) {
+	err = os.Chdir(filepath.Dir(path))
+	if errorutils.CheckError(err) != nil {
+		return
+	}
+	log.Debug("Preparing to populate mod", filepath.Dir(path))
+	// Remove go.sum file to avoid checksum conflicts with the old go.sum
+	goSum := filepath.Join(filepath.Dir(path), "go.sum")
+	exists, err := fileutils.IsFileExists(goSum, false)
+	if err != nil {
+		return
+	}
+
+	if exists {
+		err = os.Remove(goSum)
+		if errorutils.CheckError(err) != nil {
+			return
+		}
+	}
+
+	if shouldRunGoModCommand {
+		// Running go mod tidy command
+		err = golang.RunGoModTidy()
+		if err != nil {
+			return
+		}
+	}
+	if shouldRunGoGraph {
+		// Running go mod graph command
+		output, err = golang.GetDependenciesGraph()
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+// Downloads the mod file from Artifactory to the Go cache
+func downloadModFileFromArtifactoryToLocalCache(cachePath, targetRepo, name, version string, details *config.ArtifactoryDetails, client *httpclient.HttpClient) string {
+	pathToModuleCache := filepath.Join(cachePath, name, "@v")
+	dirExists, err := fileutils.IsDirExists(pathToModuleCache, false)
+	if err != nil {
+		log.Debug("Received and error:", err)
+		return ""
+	}
+
+	if dirExists {
+		url := details.Url + "api/go/" + targetRepo + "/" + name + "/@v/" + version + ".mod"
+		log.Debug("Downloading mod file from Artifactory:", url)
+		auth, err := details.CreateArtAuthConfig()
+		if err != nil {
+			log.Debug("Received and error:", err)
+			return ""
+		}
+		downloadFileDetails := &httpclient.DownloadFileDetails{
+			FileName: version + ".mod",
+			// Artifactory URL
+			DownloadPath:  url,
+			LocalPath:     pathToModuleCache,
+			LocalFileName: version + ".mod",
+		}
+		resp, err := client.DownloadFile(downloadFileDetails, "", auth.CreateHttpClientDetails(), 3, false)
+		if err != nil {
+			log.Debug("Received and error:", err)
+			return ""
+		}
+
+		log.Debug(fmt.Sprintf("Received %d from Artifactory %s", resp.StatusCode, url))
```

Comment:
I'm not sure how informative this debug message will be, because it doesn't include any context.
I suggest you consider removing it, or if you think it is important, try adding some meaningful context to it.